gems/aws-sdk-core/lib/aws-sdk-core/plugins/regional_endpoint.rb

```diff
@@ -150,7 +150,18 @@ def resolve_endpoint(cfg)
           # that a custom endpoint has NOT been configured by the user
           cfg.override_config(:regional_endpoint, true)
```

Comment:
What about using cfg.override_config for endpoint to be the value of resolve legacy endpoint?